File: backend/WebStore.API/WebStore.Domain/Entities/Wishlist.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WebStore.Domain.Entities;

public sealed class Wishlist
{
    public string? Id { get; set; } = Guid.NewGuid().ToString();
    public string? UserId { get; set; }
    
    [ForeignKey("UserId")]
    public User? User {get; set;}
    public List<WishlistItem> WishlistItems { get; set; } = [];

    public void UpdateWishlist(Wishlist wishlist)
    {
        wishlist.Id = Id;
        wishlist.UserId = UserId;
        wishlist.WishlistItems = WishlistItems;
    }
}

File: frontend/src/app/features/wishlist/domain/models/wishlist.ts
import { v4 as uuid } from 'uuid';
import { WishlistItem } from './wishlist-item';

export interface IWishlist {
  id: string;
  userId: string;
  wishlistItems: WishlistItem[];
}

export class Wishlist implements IWishlist {
  id: string;
  userId: string;
  wishlistItems: WishlistItem[];

  constructor() {
    this.id = uuid();
    this.userId = localStorage.getItem('userId') ?? '';
    this.wishlistItems = [];
  }
}
